### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by janicklas-ralph**
Optimizations on the modern bundle

### CODE REVIEW

File: packages/next/client/url.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { encode, decode } from 'querystring'
```
Comment (developit) [inline packages/next/client/url.js:1]: This ends up being handled by the `querystring-->qss` alias for sure, right?

File: packages/next/client/url.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+const IS_BROWSER = typeof window !== 'undefined'
```
Comment (developit) [inline packages/next/client/url.js]: Here's a thought: why don't we just make this module browser-only, and only enable the `url` and `querystring` aliases for the browser build?  We could alias `querystring-es3` to `querystring` for the server build.

File: packages/next/package.json [hunk 3]
```diff
@@ -97,6 +98,7 @@
     "tty-aware-progress": "1.0.4",
     "unfetch": "4.1.0",
     "url": "0.11.0",
+    "url-polyfill": "1.1.7",
```
Comment (developit) [inline packages/next/package.json]: remove?

File: yarn.lock [hunk 4]
```diff
@@ -13253,6 +13258,11 @@ url-parse-lax@^3.0.0:
   dependencies:
     prepend-http "^2.0.0"
 
+url-polyfill@^1.1.7:
```
Comment (developit) [inline yarn.lock]: (me too)

File: packages/next/client/url.js (hunk 1 above)
Comment (janicklas-ralph) [reply packages/next/client/url.js:1]: Yes. I'm aliasing `querystring` only in the client side webpack config

File: packages/next/client/url.js (hunk 2 above)
Comment (janicklas-ralph) [reply packages/next/client/url.js]: Okay, makes sense. I will update the changes

**Pull request merged**